### ISSUE TITLE
Mirror private channels

### ISSUE THREAD

**Issue by kpengboy** [ocf/slackbridge#52]
Slackbridge should be able to mirror private channels too, as secret channels with restricted access.
Registration with NickServ would be a prerequisite for joining the corresponding IRC channel.

**Comment by jvperrin** [comment]
This is mentioned already in #7, but it's good to have a more specific issue for it.

**Comment by kkuehler** [comment]
Yes, this is possible. We could receive a [member_joined_channel](https://api.slack.com/events/member_joined_channel) event. A channel object has an `is_private` property. When the bridge is invited to a private channel it could spin up an invite only IRC channel and add all corresponding slack bots.
